src/panels/media-browser/ha-bar-media-player.ts

```diff
@@ -0,0 +1,416 @@
+import type { HomeAssistant } from "../../types";
+import "../lovelace/components/hui-marquee";
+import { showSelectMediaPlayerDialog } from "./show-select-media-source-dialog";
+
+@customElement("ha-bar-media-player")
+class BarMediaPlayer extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public entityId!: string;
+
+  @property({ type: Boolean, reflect: true })
+  public narrow!: boolean;
+
+  @query("mwc-linear-progress") private _progressBar?: LinearProgress;
+
+  @query("#CurrentProgress") private _currentProgress?: HTMLElement;
+
+  @state() private _marqueeActive = false;
+
+  private _progressInterval?: number;
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+
+    const stateObj = this._stateObj;
+
+    if (!stateObj) {
+      return;
+    }
+
+    if (
+      !this._progressInterval &&
+      this._showProgressBar &&
+      stateObj.state === "playing"
+    ) {
+      this._progressInterval = window.setInterval(
+        () => this._updateProgressBar(),
+        1000
+      );
+    }
+  }
+
+  public disconnectedCallback(): void {
+    if (this._progressInterval) {
+      clearInterval(this._progressInterval);
+      this._progressInterval = undefined;
+    }
+  }
+
+  public willUpdate(changedProps: PropertyValues): void {
+    super.willUpdate(changedProps);
+    if (!this.hass || !changedProps.has("hass")) {
+      return;
+    }
+
+    const stateObj = this._stateObj;
+
+    if (!stateObj) {
+      if (this._progressInterval) {
+        clearInterval(this._progressInterval);
+        this._progressInterval = undefined;
+      }
+    }
+  }
+
+  protected render(): TemplateResult {
+    const choosePlayerElement = html`
+      <div class="choose-player">
+        ${!this.narrow
+          ? html`
+              <mwc-select
+                .label=${this.hass.localize(
+                  "ui.panel.config.automation.editor.conditions.type_select"
+                )}
+                .value=${this.entityId}
+                naturalMenuWidth
+              >
+                <mwc-list-item
+                  .player=${BROWSER_PLAYER}
+                  @click=${this._selectPlayer}
+                  >${this.hass.localize(
+                    "ui.components.media-browser.web-browser"
+                  )}</mwc-list-item
+                >
+                ${this._mediaPlayerEntities.map(
+                  (source) => html`
+                    <mwc-list-item
+                      ?selected=${source.entity_id === this.entityId}
+                      .disabled=${UNAVAILABLE_STATES.includes(source.state)}
+                      .player=${source.entity_id}
+                      @click=${this._selectPlayer}
+                      >${computeStateName(source)}</mwc-list-item
+                    >
+                  `
+                )}
+              </mwc-select>
+            `
+          : html`
+              <ha-icon-button
```

Comment:
Should we use a `<ha-menu-button>` ?

Reply:
It shows a dialog on mobile which with many entries may be better. Either way works for me

Reply:
On mobile, having it pop up where your finger is, is better.